### DISCUSSION TITLE
how to make app.prepare faster or make nextjs compile faster when using custom server?

### DISCUSSION THREAD

**Discussion by p32929** [vercel/next.js#17965] [Help]
## Describe the bug
Not sure if its a bug or something else. But when using NextJS without any custom server, it compiles the pages pretty fast. But when using a custom server ( like express ), the app.prepare promise takes about 10-20 seconds to resolve.
Is there any way to make it faster? or is there any way so that it can automatically determine, whether react code is changed or server code is changed, and it will run/re-run only the necessary functions/portion of codes?

## To Reproduce
I'm using this configuration for custom express server using NextJS: https://github.com/p32929/next-p32929/blob/master/server/apis/server.js
( Using this boilerplate: https://github.com/p32929/next-p32929/ )

When changing anything whether it be in the React code or in the express-sevrer ( in the server.js file ) code, it takes about 10-20 seconds to compile the pages.

## Expected behavior
I wish it was faster like working in Normal CRA app or like when running `next` instead of `nodemon server/apis/server.js`

## System information
- OS: Windows 10 64bit ( 12 GB RAM )
- Browser: Chrome
- Version of Next.js: 9.5.2
- Version of Node.js: 12.18.2

## Additional context
I tried Googling the problem. but found no solution that works

**Comment by timneutkens** [comment]
`next` / `next dev` runs the exact same methods as a custom server so there shouldn't be a difference 🤔
